### PR TITLE
Fix enhanced EA Mercury telemetry prices are 0 when task result is not float64

### PR DESCRIPTION
**Pull request by george-dorin**
Use utils.ToDecimal instead of casting to float64

### CODE REVIEW

File: core/services/ocrcommon/telemetry.go [hunk 1]
```diff
@@ -365,41 +365,41 @@ func (e *EnhancedTelemetryService[T]) getPricesFromResults(startTask pipeline.Ta
 		if benchmarkPriceTask.Result.Error != nil {
 			e.lggr.Warnw(fmt.Sprintf("got error for enhanced EA telemetry benchmark price, job %d, id %s: %s", e.job.ID, benchmarkPriceTask.Task.DotID(), benchmarkPriceTask.Result.Error), "err", benchmarkPriceTask.Result.Error)
 		} else {
-			benchmarkPrice, ok = benchmarkPriceTask.Result.Value.(float64)
-			if !ok {
-				e.lggr.Warnf("cannot parse enhanced EA telemetry benchmark price, job %d, id %s (expected float64, got type: %T)", e.job.ID, benchmarkPriceTask.Task.DotID(), benchmarkPriceTask.Result.Value)
+			benchmarkPrice, err = getResultFloat64(benchmarkPriceTask)
+			if err != nil {
+				e.lggr.Warnf(fmt.Sprintf("cannot parse enhanced EA telemetry benchmark price, job %d, id %s", e.job.ID, benchmarkPriceTask.Task.DotID()), "err", err)
```
Comment (samsondav) [inline core/services/ocrcommon/telemetry.go:370]: This won't lot as you expect. Needs to be:
```suggestion
				e.lggr.Warnw(fmt.Sprintf("cannot parse enhanced EA telemetry benchmark price, job %d, id %s", e.job.ID, benchmarkPriceTask.Task.DotID()), "err", err)
```

File: core/services/ocrcommon/telemetry.go [hunk 2]
```diff
@@ -365,41 +365,41 @@ func (e *EnhancedTelemetryService[T]) getPricesFromResults(startTask pipeline.Ta
...
 	if bidTask.Task.Type() == pipeline.TaskTypeJSONParse {
 		if bidTask.Result.Error != nil {
 			e.lggr.Warnw(fmt.Sprintf("got error for enhanced EA telemetry bid price, job %d, id %s: %s", e.job.ID, bidTask.Task.DotID(), bidTask.Result.Error), "err", bidTask.Result.Error)
 		} else {
-			bidPrice, ok = bidTask.Result.Value.(float64)
-			if !ok {
-				e.lggr.Warnf("cannot parse enhanced EA telemetry bid price, job %d, id %s (expected float64, got type: %T)", e.job.ID, bidTask.Task.DotID(), bidTask.Result.Value)
+			bidPrice, err = getResultFloat64(bidTask)
+			if err != nil {
+				e.lggr.Warnf(fmt.Sprintf("cannot parse enhanced EA telemetry bid price, job %d, id %s", e.job.ID, bidTask.Task.DotID()), "err", err)
```
Comment (samsondav) [inline core/services/ocrcommon/telemetry.go:386]:
```suggestion
				e.lggr.Warnw(fmt.Sprintf("cannot parse enhanced EA telemetry bid price, job %d, id %s", e.job.ID, bidTask.Task.DotID()), "err", err)
```

File: core/services/ocrcommon/telemetry.go [hunk 3]
```diff
@@ -365,41 +365,41 @@ func (e *EnhancedTelemetryService[T]) getPricesFromResults(startTask pipeline.Ta
...
 	if askTask.Task.Type() == pipeline.TaskTypeJSONParse {
 		if bidTask.Result.Error != nil {
 			e.lggr.Warnw(fmt.Sprintf("got error for enhanced EA telemetry ask price, job %d, id %s: %s", e.job.ID, askTask.Task.DotID(), askTask.Result.Error), "err", askTask.Result.Error)
 		} else {
-			askPrice, ok = askTask.Result.Value.(float64)
-			if !ok {
-				e.lggr.Warnf("cannot parse enhanced EA telemetry ask price, job %d, id %s (expected float64, got type: %T)", e.job.ID, askTask.Task.DotID(), askTask.Result.Value)
+			askPrice, err = getResultFloat64(askTask)
+			if err != nil {
+				e.lggr.Warnf(fmt.Sprintf("cannot parse enhanced EA telemetry ask price, job %d, id %s", e.job.ID, askTask.Task.DotID()), "err", err)
```
Comment (samsondav) [inline core/services/ocrcommon/telemetry.go:402]:
```suggestion
				e.lggr.Warnw(fmt.Sprintf("cannot parse enhanced EA telemetry ask price, job %d, id %s", e.job.ID, askTask.Task.DotID()), "err", err)
```

File: core/services/ocrcommon/telemetry.go [hunk 4]
```diff
@@ -431,3 +431,13 @@ func EnqueueEnhancedTelem[T EnhancedTelemetryData | EnhancedTelemetryMercuryData
...
 	}
 }
+
+// getResultFloat64 will check the result type and force it to float64 or returns an error if the conversion cannot be made
+func getResultFloat64(task *pipeline.TaskRunResult) (float64, error) {
+	result, err := utils.ToDecimal(task.Result.Value)
+	if err != nil {
+		return 0, err
+	}
+	resultFloat64, _ := result.Float64()
```
Comment (samsondav) [inline core/services/ocrcommon/telemetry.go:441]: Whats the second return value here?

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/services/ocrcommon/telemetry.go (hunk 4 above)
Comment (george-dorin) [reply core/services/ocrcommon/telemetry.go:441]: It's a `bool` that indicates whether the float is exactly represented

File: core/services/ocrcommon/telemetry.go (hunk 3 above)
Comment (george-dorin) [reply core/services/ocrcommon/telemetry.go]: Fixed in [`eba6216` (#10828)](https://github.com/smartcontractkit/chainlink/pull/10828/commits/eba621657f7d96acf33f7ee9596150ba8ea4104d)

File: core/services/ocrcommon/telemetry.go (hunk 2 above)
Comment (george-dorin) [reply core/services/ocrcommon/telemetry.go]: Fixed in [`eba6216` (#10828)](https://github.com/smartcontractkit/chainlink/pull/10828/commits/eba621657f7d96acf33f7ee9596150ba8ea4104d)

File: core/services/ocrcommon/telemetry.go (hunk 1 above)
Comment (george-dorin) [reply core/services/ocrcommon/telemetry.go]: Fixed in [`eba6216` (#10828)](https://github.com/smartcontractkit/chainlink/pull/10828/commits/eba621657f7d96acf33f7ee9596150ba8ea4104d)

**Review by samsondav — APPROVED** [review]

**Pull request merged**